File: Project/PassFailExamResult.cpp
#include "PassFailExamResult.hpp"


PassFailExamResult::PassFailExamResult(bool passed) {
	m_passed = passed;
}

std::string PassFailExamResult::ToString() const {
	return m_passed ? "Зачёт" : "Незачёт";
}

std::string PassFailExamResult::ToStringCompact() const {
	return m_passed ? "Зач" : "Нез";
}

int PassFailExamResult::ToPercent() const {
	return m_passed ? MAX_PERCENT : 0;
}

bool PassFailExamResult::IsPassed() const {
	return m_passed;
}

int PassFailExamResult::ToScore() const {
	return ToPercent();
}

File: ProjectCS/PassFailExamResult.cs
using System;

namespace SessionResultsCS
{
    // Результат зачёта
    public class PassFailExamResult : AttestationResult
    {
        // Получен ли зачёт
        public bool Passed { get; }


        public PassFailExamResult(bool passed)
        {
            Passed = passed;
        }


        public override string ToString()
        {
            return Passed ? "Зачёт" : "Незачёт";
        }

        public override string ToStringCompact()
        {
            return Passed ? "Зач" : "Нез";
        }

        public override int ToPercent()
        {
            return Passed ? MAX_PERCENT : 0;
        }

        public override bool IsPassed()
        {
            return Passed;
        }

        public override int ToScore()
        {
            return ToPercent();
        }
    }
}
